backend/corpora/lambdas/api/v1/dataset.py

```diff
@@ -48,3 +54,27 @@ def get_status(dataset_uuid: str, user: str):
     for remove in ["dataset", "created_at", "updated_at"]:
         status.pop(remove)
     return make_response(jsonify(status), 200)
+
+
+@db_session()
+def delete_dataset(dataset_uuid: str, user: str):
+    """
+    Cancels an inprogress upload.
+    """
+    dataset = Dataset.get(dataset_uuid)
+    if not dataset:
+        raise NotFoundHTTPException(f"'dataset/{dataset_uuid}' not found.")
```

Comment:
For systems that require auth for some resource, you should return ForbiddenHTTPException both when an item is to found and when the user is not authorized. This prevents users from probing the system for uuids.

```suggestion
         raise ForbiddenHTTPException()
```